Content.Shared/Silicons/Borgs/SharedBorgSystem.cs

```diff
@@ -32,13 +37,32 @@ public override void Initialize()
         SubscribeLocalEvent<BorgChassisComponent, ItemSlotEjectAttemptEvent>(OnItemSlotEjectAttempt);
         SubscribeLocalEvent<BorgChassisComponent, EntInsertedIntoContainerMessage>(OnInserted);
         SubscribeLocalEvent<BorgChassisComponent, EntRemovedFromContainerMessage>(OnRemoved);
+        SubscribeLocalEvent<BorgChassisComponent, PlayerSpawnCompleteEvent>(OnSpawn);
         SubscribeLocalEvent<BorgChassisComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovementSpeedModifiers);
         SubscribeLocalEvent<BorgChassisComponent, ActivatableUIOpenAttemptEvent>(OnUIOpenAttempt);
+
         SubscribeLocalEvent<TryGetIdentityShortInfoEvent>(OnTryGetIdentityShortInfo);
 
         InitializeRelay();
     }
 
+    private void OnSpawn(EntityUid uid, BorgChassisComponent component, PlayerSpawnCompleteEvent args)
+    {
+        RoundStartSilicon(args.Mob);
+    }
+
+    /// <summary>
+    /// Borgs and AI that enter the game through the normal StationSpawningSystem get their Silicon mindrole here
+    /// </summary>
+    public void RoundStartSilicon(EntityUid uid)
```

Comment:
If someone would add derelict or syndicate borgs at roundstart, would this get updated in the correct order?

Reply:
It would not, this would override the Altered Silicon role. 
Altered Silicon is added when the mind is added to the entity, which happens before PlayerSpawnComplete which is when RoundStartSilicon is performed